scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -257,6 +258,17 @@ object Execution {
   def withConfig[T](ex: Execution[T])(c: Config => Config): Execution[T] =
     TransformedConfig(ex, c)
 
+  def withCached[T](path: String)(fn: CachedFile => Execution[T]): Execution[T] = {
```

Comment:
Add a comment here that shows how to use this with multiple files. Consider adding a reminder comment about `CachedFile` expectations i.e. if you're using `Hdfs` mode, the `CachedFile` path will only be valid in the `TypedPipe`/`KeyedList` closures (right?).